arangod/RocksDBEngine/RocksDBMetaCollection.cpp

```diff
@@ -884,3 +881,52 @@ int RocksDBMetaCollection::doLock(double timeout, AccessMode::Type mode) {
     }
   }
 }
+
+bool RocksDBMetaCollection::haveBufferedOperations() const {
+  if (!_logicalCollection.useSyncByRevision()) {
+    return false;
+  }
+
+  std::unique_lock<std::mutex> guard(_revisionBufferLock);
+
+  // have a truncate to apply
+  if (!_revisionTruncateBuffer.empty()) {
+    return true;
+  }
+
+  // have insertions to apply
+  if (!_revisionInsertBuffers.empty()) {
+    return true;
+  }
+
+  // have removals to apply
+  if (!_revisionRemovalBuffers.empty()) {
+    return true;
+  }
+
+  return false;
+}
+
+std::size_t RocksDBMetaCollection::revisionTreeDepth() const {
+  return 6;  // TODO make variable
+}
+
+std::unique_ptr<containers::RevisionTree> RocksDBMetaCollection::allocateEmptyRevisionTree() const {
+  return std::make_unique<containers::RevisionTree>(
+      revisionTreeDepth(), _logicalCollection.minRevision().id());
+}
+
+bool RocksDBMetaCollection::ensureRevisionTree() {
+  try {
+    if (_revisionTree) {
+      return true;
+    }
+    _revisionTreeCreationSeq = rocksutils::globalRocksDB()->GetLatestSequenceNumber();
+    _revisionTreeSerializedSeq = _revisionTreeCreationSeq;
+    _revisionTree = allocateEmptyRevisionTree();
+    return true;
+  } catch (...) {
+    _revisionTree.reset();  // should be unnecessary, but just in case
+    return false;
+  }
+}
```

Comment:
Looks like this method's return value is never checked, so we may simplify it to just:
```suggestion
void RocksDBMetaCollection::ensureRevisionTree() {
  if (!_revisionTree) {
    // if the allocation throws, _revisionTree` will still be a nullptr.
    _revisionTree = allocateEmptyRevisionTree();
    _revisionTreeCreationSeq = rocksutils::globalRocksDB()->GetLatestSequenceNumber();
    _revisionTreeSerializedSeq = _revisionTreeCreationSeq;
  }
}
```